workflows/cloudify_system_workflows/snapshot.py

```diff
@@ -0,0 +1,574 @@
+
+def _create(ctx, snapshot_id, config, include_metrics, include_credentials,
+            **kw):
+    tempdir = tempfile.mkdtemp('-snapshot-data')
+
+    snapshots_dir = os.path.join(
+        config.file_server_root,
+        config.file_server_snapshots_folder
+    )
+
+    if not os.path.exists(snapshots_dir):
+        os.makedirs(snapshots_dir)
+
+    metadata = {}
+
+    # files/dirs copy
+    _copy_data(tempdir, config)
+
+    # elasticsearch
+    es = _create_es_client(config)
+    has_cloudify_events = es.indices.exists(index=_EVENTS_INDEX_NAME)
+    _dump_elasticsearch(tempdir, es, ctx.execution_id, has_cloudify_events)
+
+    metadata[_M_HAS_CLOUDIFY_EVENTS] = has_cloudify_events
+
+    # influxdb
+    if include_metrics:
+        _dump_influxdb(tempdir)
+
+    # credentials
+    if include_credentials:
+        ctx.load_deployments_contexts()
+        _dump_credentials(ctx, tempdir)
+
+    # version
+    metadata[_M_VERSION] = _VERSION
+
+    # metadata
+    with open(os.path.join(tempdir, _METADATA_FILE), 'w') as f:
+        json.dump(metadata, f)
+
+    # agents
+    _dump_agents(tempdir, ctx)
+
+    # zip
+    snapshot_dir = os.path.join(snapshots_dir, snapshot_id)
+    os.makedirs(snapshot_dir)
+
+    shutil.make_archive(
+        os.path.join(snapshot_dir, snapshot_id),
+        'zip',
+        tempdir
+    )
+
+    # end
+    shutil.rmtree(tempdir)
+
+
+@workflow(system_wide=True)
+def create(ctx, snapshot_id, config, **kwargs):
+    update_status = get_rest_client().snapshots.update_status
+    config = _DictToAttributes(config)
+    try:
+        _create(ctx, snapshot_id, config, **kwargs)
+        update_status(snapshot_id, config.created_status)
+    except BaseException, e:
+        update_status(snapshot_id, config.failed_status, str(e))
+        raise
+
+
+def _dump_elasticsearch(tempdir, es, execution_id, has_cloudify_events):
+    storage_scan = elasticsearch.helpers.scan(es, index=_STORAGE_INDEX_NAME)
+    storage_scan = _except_types(storage_scan,
+                                 'provider_context',
+                                 'snapshot')
+    storage_scan = (e for e in storage_scan if e['_id'] != execution_id)
+
+    event_scan = elasticsearch.helpers.scan(
+        es,
+        index=_EVENTS_INDEX_NAME if has_cloudify_events else 'logstash-*'
+    )
+
+    with open(os.path.join(tempdir, _ELASTICSEARCH), 'w') as f:
+        for item in itertools.chain(storage_scan, event_scan):
+            f.write(json.dumps(item) + '\n')
+
+
+def _dump_influxdb(tempdir):
+    influxdb_file = os.path.join(tempdir, _INFLUXDB)
+    influxdb_temp_file = influxdb_file + '.temp'
+    subprocess.call(_INFLUXDB_DUMP_CMD.format(influxdb_temp_file), shell=True)
+    with open(influxdb_temp_file, 'r') as f, open(influxdb_file, 'w') as g:
+        for obj in _get_json_objects(f):
+            g.write(obj + '\n')
+
+    os.remove(influxdb_temp_file)
+
+
+def _is_compute(node):
+    return 'cloudify.nodes.Compute' in node.type_hierarchy
```

Comment:
`cloudify.nodes.Compute` can be imported from `cloudify.constants.COMPUTE_NODE_TYPE`